generator/ServiceClientGeneratorLib/Generators/Marshallers/RestXmlResponseUnmarshaller.tt

```diff
@@ -256,4 +257,29 @@ namespace <#=this.Config.Namespace #>.Model.Internal.MarshallTransformations
     this.AddResponseSingletonMethod();
 #>
     }
-}
\ No newline at end of file
+}
+<#+
+    /* The rules for determining the marshallName for xml services is different than json services
+        *  1. Use the member locationName if the value doesn't match the member's name
+        2. Use the value of the locationName on the member's target if present
+        3. Use the name of the member's target
+    */
+    protected string DetermineXmlMarshallName(Member member)
+    {
+        var locationName = member.data[ServiceModel.LocationNameKey];
+        if(locationName == null)
+            return member.MarshallName;
+        if (!string.Equals(locationName.ToString(), member.ModeledName, StringComparison.Ordinal))
+        {
+            return locationName.ToString();
+        }
+        var memberTarget = member.Shape.data[ServiceModel.LocationNameKey];
+        if(member.Shape.data != null && memberTarget != null)
```

Comment:
`member.Shape.data != null` doesn't seem needed as you would have referenced a location in a null json object previously and that would have failed. Does the check need to be moved or removed?

Reply:
you're right, Member.Shape.data would never actually return null I removed it